components/nexusoperations/events.go

```diff
@@ -112,10 +112,18 @@ func (d CompletedEventDefinition) IsWorkflowTaskTrigger() bool {
 
 func (d CompletedEventDefinition) Apply(root *hsm.Node, event *historypb.HistoryEvent) error {
 	return transitionOperation(root, event, func(node *hsm.Node, o Operation) (hsm.TransitionOutput, error) {
-		return TransitionSucceeded.Apply(o, EventSucceeded{
+		output, err := TransitionSucceeded.Apply(o, EventSucceeded{
 			Time: event.EventTime.AsTime(),
 			Node: node,
 		})
+		if err != nil {
+			return output, err
+		}
+
+		if err := root.DeleteChild(node.Key); err != nil {
```

Comment:
Technically you'll want to separate this into two operations, a transition, followed by a deletion. Move the DeleteChild call to after the transitionOperation call.

Reply:
There's also a boolean on the operation state saying whether it should be deleted on completion but we can get rid of that and just delete here always.

Reply:
Done